src/Domain/CoordinateMaps/TimeDependent/Translation.cpp

```diff
@@ -84,25 +106,83 @@ std::optional<std::array<double, Dim>> Translation<Dim>::inverse(
              << function_of_time.size()
              << ") does not match the dimension of the translation map (" << Dim
              << ").");
-  double radial_function_value = 1.0;
-  if (f_of_r_ != nullptr) {
-    const double root = root_finder(target_coords - center_, function_of_time);
-    radial_function_value = (*f_of_r_)(root);
-  }
-  for (size_t i = 0; i < Dim; i++) {
-    gsl::at(result, i) -= function_of_time[i] * radial_function_value;
+  // If an inner radius is not specified then take the inverse of the
+  // MathFunction specific translation.
+  if (!inner_radius_.has_value()) {
+    double radial_function_value = 1.0;
+    // If a MathFunction is specified, do a root find.
+    if (f_of_r_ != nullptr) {
+      const double root =
+          root_finder(target_coords - center_, function_of_time);
+      radial_function_value = (*f_of_r_)(root);
+    }
+    for (size_t i = 0; i < Dim; i++) {
+      gsl::at(result, i) -= function_of_time[i] * radial_function_value;
+    }
+    return result;
+  } else {
+    // Section for taking the inverse of the piecewise translation
+    auto target_radius = magnitude(target_coords);
+    double non_translated_radius = 0.;
+    for (size_t i = 0; i < Dim; i++) {
+      non_translated_radius +=
+          square(gsl::at(target_coords, i) - function_of_time[i]);
+    }
+    non_translated_radius = sqrt(non_translated_radius);
+    // Assumes no translation if the target radius is greater than the
+    // outer radius.
+    if (target_radius >= outer_radius_.value()) {
+      return result;
+    } else if (non_translated_radius <= inner_radius_.value()) {
+      for (size_t i = 0; i < Dim; i++) {
+        gsl::at(result, i) -= function_of_time[i];
+      }
+    } else {
+      // We need to solve a quadratic of the form w^2a + wb + c where
+      // a = T(t)^2 - (R_{out} - R_{in})^2,
+      // b = 2(R_{out}(R_{out} - R_{in}) - T(t)\vec{\bar{\xi}}), and
+      // c = \vec{\bar{\xi}}^2 - R_{out}^2
```

Comment:
`c` is different than in the docs in the hpp. Which is correct?

Reply:
This c is correct, will correct the hpp